packages/now-node/src/index.ts

```diff
@@ -470,31 +471,47 @@ export async function startDevServer(
 async function doTypeCheck({
   entrypoint,
   workPath,
+  meta = {},
 }: StartDevServerOptions): Promise<void> {
-  // In order to type-check a single file, a temporary tsconfig
+  const { devCacheDir = join(workPath, '.vercel', 'cache') } = meta;
+
+  // In order to type-check a single file, a standalone tsconfig
   // file needs to be created that inherits from the base one :(
   // See: https://stackoverflow.com/a/44748041/376773
-  const id = Math.random()
-    .toString(32)
-    .substring(2);
-  const tempConfigName = `.tsconfig-${id}.json`;
   const projectTsConfig = await walkParentDirs({
     base: workPath,
     start: join(workPath, dirname(entrypoint)),
     filename: 'tsconfig.json',
   });
+  const sha = createHash('sha256')
+    .update(entrypoint)
+    .update(projectTsConfig || '')
+    .digest('hex');
+  const tempConfigPath = join(devCacheDir, `tsconfig-${sha}.json`);
```

Comment:
```suggestion
  const tempConfigPath = join(devCacheDir, `tsconfig-${sha}.json`);
```

```suggestion
  const tempConfigPath = join(devCacheDir, 'now-node', entrypoint, `tsconfig-${sha}.json`);
```

Will this file differ for each entrypoint? Also it should probably be under `node` to avoid overrwriting other builder/runtime cache. The other are using `now-node` and `now-python`

Reply:
The filename is different for each entrypoint (see how the `sha` is generated). I'm not sure the subdir is strictly necessary in this case, as the sha should be unique enough to avoid collisions.

If anything, I'd prefer to add `vercel-node` as part of how the sha is generated, to avoid having to `mkdir -p` which we don't have a module for in `@vercel/node`.

Reply:
I think I missed the opportunity for the CLI to create a cache directory based on the Runtime name, that would have been the best solution. The problem here is that someone is going to run the latest `@vercel/node` on an older version of the CLI which didn't `mkdirp()` yet. So unless we pin the builder versions we are kinda stuck with creating the cache directory in the builder itself.

Reply:
Updated to use `mkdirp-promise` module and use subdirectories. As a consequence, there is no need for SHA hashing of the entrypoint filename anymore, which simplifies the code a bit.